### PR TITLE
Work bsc#1103199 and bsc#1112109 around

### PR DESCRIPTION
**Pull request by rwx788**
We have a bug, which interferes with usage of serial console. On qemu
backend it's enough to wait for welcome message on tty, which means that
service is running. On Xen HVM service gets restarted constantly, so the
only possible workaround is to stop it ASAP.

See [poo#42359](https://progress.opensuse.org/issues/42359).

#### Verification runs
[default ppc64le](http://g226.suse.de/tests/2825)
[default 64bit](http://g226.suse.de/tests/2846)
[opensuse gnome](http://g226.suse.de/tests/2849)

### CODE REVIEW

File: lib/main_common.pm [hunk 1]
```diff
@@ -18,7 +18,27 @@ use testapi qw(check_var get_var get_required_var set_var check_var_array diag);
 use autotest;
 use utils;
 use version_utils qw(
-  is_hyperv is_hyperv_in_gui is_jeos is_gnome_next is_krypton_argon is_leap is_opensuse is_sle is_sles4sap is_sles4sap_standard sle_version_at_least is_desktop_installed is_installcheck is_rescuesystem is_staging is_tumbleweed is_virtualization_server is_caasp is_upgrade
+  is_caasp
```
Comment (foursixnine) [inline lib/main_common.pm]: \o/

File: lib/main_common.pm [hunk 2]
```diff
@@ -17,9 +17,7 @@ use Exporter;
 use testapi qw(check_var get_var get_required_var set_var check_var_array diag);
 use autotest;
 use utils;
-use version_utils qw(
-  is_hyperv is_hyperv_in_gui is_jeos is_gnome_next is_krypton_argon is_leap is_opensuse is_sle is_sles4sap is_sles4sap_standard sle_version_at_least is_desktop_installed is_installcheck is_rescuesystem is_staging is_tumbleweed is_virtualization_server is_caasp is_upgrade
-);
+use version_utils qw(:VERSION :BACKEND :SCENARIO);
```
Comment (foursixnine) [inline lib/main_common.pm]: Great change @rwx788

**Comment by foursixnine** [comment]
Let's do this!

File: lib/opensusebasetest.pm [hunk 3]
```diff
@@ -456,6 +457,8 @@ sub wait_boot {
 
     # on s390x svirt encryption is unlocked with workaround_type_encrypted_passphrase before here
     unlock_if_encrypted if !get_var('S390_ZKVM');
+    # On remote backends we sync on Welcome message already where possible, mitigating bsc#1112109
+    wait_serial(get_login_message()) unless is_remote_backend;
```
Comment (okurz) [inline lib/opensusebasetest.pm]: Oh, so this is why we have in so many tests this failing step. The return value is ignored. Does this wait_serial actually work in your local tests?

File: tests/console/consoletest_setup.pm [hunk 4]
```diff
@@ -57,8 +61,6 @@ sub run {
         assert_script_run 'less /var/log/YaST2/y2log*|grep "Automatic DHCP configuration not started - an interface is already configured"';
     }
 
-    # Stop serial-getty on serial console to avoid serial output pollution with login prompt
-    systemctl "stop serial-getty\@$testapi::serialdev";
     # Mask if is qemu backend as use serial in remote installations e.g. during reboot
     systemctl "mask serial-getty\@$testapi::serialdev" if check_var('BACKEND', 'qemu');
```
Comment (okurz) [inline tests/console/consoletest_setup.pm]: Why not also move this call?

File: lib/opensusebasetest.pm (hunk 3 above)
Comment (rwx788) [reply lib/opensusebasetest.pm]: Now I have actually noticed that it failed on ppc64. I will create new PR to remove this one and addressing you comment below, as reverting export changes will create many conflicts.

File: tests/console/consoletest_setup.pm (hunk 4 above)
Comment (rwx788) [reply tests/console/consoletest_setup.pm]: It's already done: https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/9c0cca4be819767949452be797692de7d9671b4b

**Pull request merged**